### PR TITLE
Updated OFP commands

### PR DESCRIPTION
**Pull request by rmcd-mscb**
Hi all, I've made changes to ofp and .yml  files to reflect recent changes in the ofp.  I tagged an new version of the ofp v0.1.2.  It has a new driver python script that has new options.  I added ENV variables START_DATE and END_DATE for the new options.  I've propagated those variables through but I'm unsure whether they should be in quotes or not?, and if in particular if they are defined correctly in docker-compose.yml.  Please advise.  

Steve and I are working on a python controller script.  My plan is to call this container from python as below.  This is probably not the place for this but I'd also appreciate some comments on how to call the ofp from python in a controller script if you have some thoughts: 

sformat = "%Y-%m-%d"
str_start_pull_date = start_pull_date.strftime(sformat)
str_end_pull_date = end_pull_date.strftime(sformat)
# START_DATE and END_DATE are ENV variables in nhmusgs-ofp Docker file we over-ride with -e option
ofp_docker_cmd = f"docker run ofp -e START_DATE={str_start_pull_date} END_DATE={str_end_pull_date}"
with open("ofp_log.log", "a") as output:
   subprocess.call(ofp_docker_cmd, stdout=output, stderr=output)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@ services:
     container_name: ofp
     environment:
       - NUMDAYS=${NUMDAYS-60}
+      - START_DATE = 2019_01_01
```
Comment (isuftin) [inline docker-compose.yml]: You may consider removing the spaces here to make the formatting consistent

File: nhmusgs-ofp/ofp [hunk 2]
```diff
@@ -1,3 +1,4 @@
 #!/bin/bash
 
-/opt/conda/bin/python -u onhm-fetcher-parser/pkg/onhm_fp_script.py -d "$NUMDAYS" -i /var/lib/nhm/ofp/Data -o /var/lib/nhm/ofp/Output
+#/opt/conda/bin/python -u onhm-fetcher-parser/pkg/onhm_fp_script.py -d "$NUMDAYS" -i /var/lib/nhm/ofp/Data -o /var/lib/nhm/ofp/Output
```
Comment (isuftin) [inline nhmusgs-ofp/ofp]: Remove this commented out line. If you do need this line for reference, it is in source control and you can see it by looking back through history

File: nhmusgs-ofp/ofp [hunk 3]
```diff
@@ -1,3 +1,4 @@
...
+/opt/conda/bin/python -u onhm-fetcher-parser/pkg/climate_etl.py -t date -p "$START_DATE" "$END_DATE"  -i /var/lib/nhm/ofp/Data -o /var/lib/nhm/ofp/Output -w var/lib/nhm/ofp/Data/weights.csv
```
Comment (isuftin) [inline nhmusgs-ofp/ofp]: Did you want to prepend a `/` the the front of the path in the `-w` flag here?

File: docker-compose.yml (hunk 1 above)
Comment (isuftin) [reply docker-compose.yml]: You also may consider having these be fallback values if the START_DATE and END_DATE variables aren't set by the caller when calling Docker Compose

https://docs.docker.com/compose/compose-file/#variable-substitution

So something like

```yml
- START_DATE=${START_DATE-2019_01_01}
- END_DATE=${END_DATE-2019_01_02}
```

File: nhmusgs-ofp/Dockerfile [hunk 4]
```diff
@@ -24,6 +24,9 @@ USER $USER
 WORKDIR /home/$USER
 
 ENV NUMDAYS=60
+ENV START_DATE = 2019_01_01
```
Comment (isuftin) [inline nhmusgs-ofp/Dockerfile]: Remove spaces here

**Comment by isuftin** [comment]
> I've propagated those variables through but I'm unsure whether they should be in quotes or not?

I think quotes should probably be fine where you're doing it, if you're referring to the python commandline. 

> and if in particular if they are defined correctly in docker-compose.yml. Please advise.

I'm beginning to think that maybe having the NUMDAYS, START_DATE and END_DATE environment vars should be relegated to `docker-compose-testing.yml` and moved out of the regular `docker-compose.yml`? That can be done in another PR but I think because those are already default environment values set in the Dockerfile anyway, not having them in `docker-compose.yml` also works but if you want to play with the values, throwing them into `docker-compose-testing.yml`, which end-users/devs are supposed to mess around with might be the best bet?

```python
sformat = "%Y-%m-%d"
str_start_pull_date = start_pull_date.strftime(sformat)
str_end_pull_date = end_pull_date.strftime(sformat)
ofp_docker_cmd = f"docker run ofp -e START_DATE={str_start_pull_date} END_DATE={str_end_pull_date}"
with open("ofp_log.log", "a") as output:
subprocess.call(ofp_docker_cmd, stdout=output, stderr=output)
```

First, I thought you were using underscores in your date values, not dashes? Should sformat be `"%Y_%m_%d"` ?

For the `docker run` portion, you will want to add a `--rm` flag to clean up the exited container or else you will have a lot of dead containers lying around on the system.

Everything else here seems good.

**Comment by isuftin** [comment]
You also may consider the Docker SDK for Python to control your containers - https://docker-py.readthedocs.io/en/stable/

File: docker-compose.yml (hunk 1 above)
Comment (rmcd-mscb) [reply docker-compose.yml]: Thanks Ivan - accept suggestion

File: nhmusgs-ofp/ofp (hunk 3 above)
Comment (rmcd-mscb) [reply nhmusgs-ofp/ofp]: yes - thanks

File: docker-compose.yml (hunk 1 above)
Comment (rmcd-mscb) [reply docker-compose.yml]: Thanks Ivan - changed to your suggestion

File: nhmusgs-ofp/ofp (hunk 2 above)
Comment (rmcd-mscb) [reply nhmusgs-ofp/ofp]: Yes - changed

File: nhmusgs-ofp/Dockerfile (hunk 4 above)
Comment (rmcd-mscb) [reply nhmusgs-ofp/Dockerfile]: Removed spaces and changed date format to use dashes not underscore

**Comment by ashalper-usgs** [comment]
@rmcd-mscb, here's what happens when I run `compose-test.sh` on my set-up:
```
Running ofp...
starting Script
instantiated
/home/nhm
/var/lib/nhm/ofp/Data
/usr/local/src/onhm-fetcher-parser/pkg/fponhm.py:145: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  self.gdf = pd.concat([gpd.read_file(f) for f in filenames]).pipe(gpd.GeoDataFrame)
['nhru_14.shp', 'nhru_15.shp', 'nhru_16.shp', 'nhru_17.shp', 'nhru_11.shp', 'nhru_12.shp', 'nhru_10U.shp', 'nhru_01.shp', 'nhru_02.shp', 'nhru_03.shp', 'nhru_04.shp', 'nhru_05.shp', 'nhru_06.shp', 'nhru_07.shp', 'nhru_08.shp', 'nhru_09.shp', 'nhru_13.shp', 'nhru_10L.shp', 'nhru_18.shp']
  LAYER    POI_ID                                           geometry  hru_id_nat  hru_id_reg region
0   NaN  16977382  POLYGON ((-108.504389746 35.69548597500005, -1...       83029           1     14
1   NaN  16977382  POLYGON ((-108.565092577 35.84979445100004, -1...       83030           2     14
2   NaN  16978712  POLYGON ((-107.551000163 35.94484814700007, -1...       83031           3     14
3   NaN  16978708  POLYGON ((-107.682940326 36.07956124200007, -1...       83032           4     14
4   NaN  16975752  (POLYGON ((-107.703448135 35.99790763500005, -...       83033           5     14
Success!
initalized

running
Traceback (most recent call last):
  File "onhm-fetcher-parser/pkg/climate_etl.py", line 128, in <module>
    main()
  File "onhm-fetcher-parser/pkg/climate_etl.py", line 110, in main
    fp.run_weights()
  File "/usr/local/src/onhm-fetcher-parser/pkg/fponhm.py", line 288, in run_weights
    wght_uofi = pd.read_csv(self.iptpath / Path(self.wghts_file))
  File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 702, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 429, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 1122, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 1853, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 387, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 705, in pandas._libs.parsers.TextReader._setup_parser_source
FileNotFoundError: [Errno 2] File b'/var/lib/nhm/ofp/Data/weights.csv' does not exist: b'/var/lib/nhm/ofp/Data/weights.csv'
```

**Comment by rmcd-mscb** [comment]
@ashalper-usgs , That error indicates the script is not finding the new weights.csv file.  I've checked that the HRU_SOURCE variable is correct (HRU_SOURCE=ftp://ftpext.usgs.gov/pub/cr/co/denver/BRR-CR/pub/rmcd/Data_hru_shp_v2.tar.gz), and the new file Data_hru_shp_v2.tar.gz exists and contains the weights.csv file, it does.  Did the nhm script run before running ofp?

> @rmcd-mscb, here's what happens when I run `compose-test.sh` on my set-up:
> 
> ```
> Running ofp...
> starting Script
> instantiated
> /home/nhm
> /var/lib/nhm/ofp/Data
> /usr/local/src/onhm-fetcher-parser/pkg/fponhm.py:145: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
> of pandas will change to not sort by default.
> 
> To accept the future behavior, pass 'sort=False'.
> 
> To retain the current behavior and silence the warning, pass 'sort=True'.
> 
>   self.gdf = pd.concat([gpd.read_file(f) for f in filenames]).pipe(gpd.GeoDataFrame)
> ['nhru_14.shp', 'nhru_15.shp', 'nhru_16.shp', 'nhru_17.shp', 'nhru_11.shp', 'nhru_12.shp', 'nhru_10U.shp', 'nhru_01.shp', 'nhru_02.shp', 'nhru_03.shp', 'nhru_04.shp', 'nhru_05.shp', 'nhru_06.shp', 'nhru_07.shp', 'nhru_08.shp', 'nhru_09.shp', 'nhru_13.shp', 'nhru_10L.shp', 'nhru_18.shp']
>   LAYER    POI_ID                                           geometry  hru_id_nat  hru_id_reg region
> 0   NaN  16977382  POLYGON ((-108.504389746 35.69548597500005, -1...       83029           1     14
> 1   NaN  16977382  POLYGON ((-108.565092577 35.84979445100004, -1...       83030           2     14
> 2   NaN  16978712  POLYGON ((-107.551000163 35.94484814700007, -1...       83031           3     14
> 3   NaN  16978708  POLYGON ((-107.682940326 36.07956124200007, -1...       83032           4     14
> 4   NaN  16975752  (POLYGON ((-107.703448135 35.99790763500005, -...       83033           5     14
> Success!
> initalized
> 
> running
> Traceback (most recent call last):
>   File "onhm-fetcher-parser/pkg/climate_etl.py", line 128, in <module>
>     main()
>   File "onhm-fetcher-parser/pkg/climate_etl.py", line 110, in main
>     fp.run_weights()
>   File "/usr/local/src/onhm-fetcher-parser/pkg/fponhm.py", line 288, in run_weights
>     wght_uofi = pd.read_csv(self.iptpath / Path(self.wghts_file))
>   File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 702, in parser_f
>     return _read(filepath_or_buffer, kwds)
>   File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 429, in _read
>     parser = TextFileReader(filepath_or_buffer, **kwds)
>   File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
>     self._make_engine(self.engine)
>   File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 1122, in _make_engine
>     self._engine = CParserWrapper(self.f, **self.options)
>   File "/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py", line 1853, in __init__
>     self._reader = parsers.TextReader(src, **kwds)
>   File "pandas/_libs/parsers.pyx", line 387, in pandas._libs.parsers.TextReader.__cinit__
>   File "pandas/_libs/parsers.pyx", line 705, in pandas._libs.parsers.TextReader._setup_parser_source
> FileNotFoundError: [Errno 2] File b'/var/lib/nhm/ofp/Data/weights.csv' does not exist: b'/var/lib/nhm/ofp/Data/weights.csv'
> ```

**Comment by ashalper-usgs** [comment]
@rmcd-mscb, I was still getting that error. Think I might've found the problem. Testing it now.

**Comment by ashalper-usgs** [comment]
One of the downloads hadn't completed on my home machine this morning. Might've gone into sleep mode overnight.

**Pull request merged**